xbmc/video/VideoDatabase.cpp

```diff
@@ -6031,9 +6032,18 @@ bool CVideoDatabase::GetSeasonsByWhere(const std::string& strBaseDir, const Filt
         pItem->GetVideoInfoTag()->m_strPath = path;
         pItem->GetVideoInfoTag()->m_strShowTitle = m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_TITLE).get_asString();
         pItem->GetVideoInfoTag()->m_strPlot = m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_PLOT).get_asString();
+
+        // tvshow premiered date - used as fallback in case no episode based
+        // premiered/first aired is available for the given season
         pItem->GetVideoInfoTag()->m_premiered.SetFromDBDate(m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_PREMIERED).get_asString());
         if (pItem->GetVideoInfoTag()->m_premiered.IsValid())
           pItem->GetVideoInfoTag()->m_iYear = pItem->GetVideoInfoTag()->m_premiered.GetYear();
+
+        // season premiered date - based on first episode airdate associated to the season
+        pItem->GetVideoInfoTag()->m_firstAired.SetFromDBDate(m_pDS->fv("aired").get_asString());
+        if (pItem->GetVideoInfoTag()->m_firstAired.IsValid())
+          pItem->GetVideoInfoTag()->m_iYear = pItem->GetVideoInfoTag()->m_firstAired.GetYear();
```

Comment:
Maybe move the logic to set the year together to make it more obvious which has preference over the other like
```
if (pItem->GetVideoInfoTag()->m_firstAired.IsValid())
  pItem->GetVideoInfoTag()->m_iYear = pItem->GetVideoInfoTag()->m_firstAired.GetYear();
else if (pItem->GetVideoInfoTag()->m_premiered.IsValid())
  pItem->GetVideoInfoTag()->m_iYear = pItem->GetVideoInfoTag()->m_premiered.GetYear();
```